codice/ddf
# DDF-6256 Upgrade vulnerable dependencies

#### What does this PR do?
Addresses OWASP findings by upgrading vulnerable dependencies and cleans up outdated OWASP suppressions.

It also enables the Jetty GzipHandler so all webapps now get gzip compression for free.

#### Who is reviewing it? 

@blen-desta @bakejeyner @garrettfreibott @stustison @emmberk 

#### Select relevant component teams: 
@codice/security   

#### How should this be tested?

Verify:
* The crypter library still works (Tink upgrade): Upload and download a resource
* The platform scheduler still works (Quartz upgrade):
  - From the Admin UI, navigate to Platform > Configuration > Platform Command Scheduler
  - Set a command, (e.g. "echo hello") and a short interval (e.g. 1 second)
  - Check the logs. The command should start running after a minute's delay
* You can load the documentation in the admin UI (Spark upgrade)
* Gzip compression is applied to all webapps. Check the Content-Encoding in the response headers in your browser devtools. Some things, like images, non-2XX responses, etc, won't be gzipped, but most plain-text responses will be. See https://www.eclipse.org/jetty/documentation/current/gzip-filter.html#gzip-filter-rules

#### Any background context you want to provide?
This is part of an ongoing effort to reduce the CVE level in DDF

#### What are the relevant tickets?
Fixes: https://github.com/codice/ddf/issues/6256

#### Screenshots


#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
